libtenzir/src/plugin.cpp

```diff
@@ -305,6 +305,19 @@ load(const std::vector<std::string>& bundled_plugins,
 
 /// Initialize loaded plugins.
 caf::error initialize(caf::actor_system_config& cfg) {
+  // If everything went well, we should have a strictly-ordered list of plugins.
+  if (auto it = std::ranges::adjacent_find(get(), std::greater_equal{});
+      it != get().end()) {
+    auto name_a = (*it)->name();
+    ++it;
+    auto name_b = (*it)->name();
+    if (name_a == name_b) {
+      TENZIR_ERROR("found multiple plugins named `{}`", name_a);
+    } else {
+      TENZIR_ERROR("unexpected plugin ordering: found `{}` before `{}`", name_a,
+                   name_b);
+    }
+  }
```

Comment:
@dominiklohmann Seems like we did not have such a check before (and there already was a name collision between the `reverse` TQL1 operator and TQL2 function). I also check the ordering here. Went for a log message instead of a hard error, because things mostly work as expected even when violating this invariant. Feel free to let me know if you think this needs changes.

Reply:
I think this should just cause a hard shutdown. It's clearly a programming error. Logging is not very visible to the app or when using systemd or Docker, so this may be too easy to overlook.

Reply:
Very true. Thanks!